ci/kokoro/cache-functions.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+readonly CACHE_KEYFILE="${KOKORO_GFILE_DIR:-/dev/shm}/build-results-service-account.json"
+
+cache_download_enabled() {
+  if [[ ! -f "${CACHE_KEYFILE}" ]]; then
+    echo "================================================================"
+    log_normal "Service account for cache access is not configured."
+    log_normal "No attempt will be made to download the cache, exit with success."
+    return 1
+  fi
+
+  if [[ "${RUNNING_CI:-}" != "yes" || (\
+    "${KOKORO_JOB_TYPE:-}" != "PRESUBMIT_GERRIT_ON_BORG" && \
+    "${KOKORO_JOB_TYPE:-}" != "PRESUBMIT_GITHUB") ]]; then
+    echo "================================================================"
+    log_normal "Cache not downloaded as this is not a PR build."
+    return 1
+  fi
+  return 0
+}
+
+cache_gcloud_cleanup() {
+  revoke_service_account_keyfile "${CACHE_KEYFILE}" || true
+  delete_gcloud_config
+}
+
+cache_download_tarball() {
+  local -r GCS_FOLDER="$1"
+  local -r DESTINATION="$2"
+  local -r FILENAME="$3"
+
+  trap cache_gcloud_cleanup EXIT
```

Comment:
I believe traps aren't local to a function, so this trap will still be set after this function exits. Therefore, I wonder if it'd be clearer to list this trap immediately after declaring the `cache_gcloud_cleanup()` function? WDYT? Is that a bad idea?